protos/transaction.proto

```diff
@@ -156,6 +158,17 @@ message Transaction {
     // The new fragments where updated rows have been moved to.
     repeated DataFragment new_fragments = 3;
   }
+  
+  // An operation that sets table metadata.
+  message SetMetadata {
+    map<string, string> metadata = 1;
+  }
+  
+  // An operation that deletes table metadata.
+  message DeleteMetadata {
+    repeated string metadata_keys = 1;
+  }
```

Comment:
I'm thinking maybe we should combine these?

```suggestion
  // An operation that modifies table metadata.
  message UpdateMetadata {
    // Values that should be inserted or updated.
    map<string, string> upsert_values = 1;
    // Keys that should be removed.
    repeated string delete_keys = 2;
  }
```

Reply:
I like it. Should this API merge carry through to the other public table metadata methods?

Reply:
I don't think it has to, but if you want you could.

Reply:
Done in 5daaf16 - I decided to still keep the methods separate in `dataset.rs` for now.